File: DataStructure/10866_덱.cpp
#include <iostream>
#include <queue>
#include <string>

using namespace std;

void solution(void) {
	int T;
	cin >> T;
	deque<int> dq;

	while (T--) {
		string cmm;
		cin >> cmm;

		if (cmm == "push_back") {
			int value;
			cin >> value;
			dq.push_back(value);
		}

		else if (cmm == "push_front") {
			int value;
			cin >> value;
			dq.push_front(value);
		}
		
		else if (cmm == "pop_front") {
			if (dq.empty())
				cout << -1 << '\n';

			else {
				cout << dq.front() << '\n';
				dq.pop_front();
			}
		}
		
		else if (cmm == "pop_back") {
			if (dq.empty())
				cout << -1 << '\n';

			else {
				cout << dq.back() << '\n';
				dq.pop_back();
			}
		}
		
		else if (cmm == "front") {
			if (dq.empty()) 
				cout << -1 << '\n';
			
			else 
				cout << dq.front() << '\n';
			
		}
		
		else if (cmm == "back") {
			if (dq.empty()) 
				cout << -1 << '\n';
			
			else 
				cout << dq.back() << '\n';
		}

		else if (cmm == "size") 
			cout << dq.size() << '\n';
		
		else if (cmm == "empty") 
			cout << dq.empty() << '\n';
	}
	return;
}

int main(void) {
	ios::sync_with_stdio(0);
	cin.tie(0);
	cout.tie(0);

	solution();
	return 0;
}

File: DataStructure/10866_덱.py
#덱
from collections import deque
#입출력
from sys import stdin

#덱 선언
dq=deque()

#입력 
input = stdin.readline

#입력
T=int(input())

for _ in range(T):
    #명령어
    comm,*n=input().split()
    
    #앞 push
    if comm == 'push_front':
        dq.appendleft(int(n[0]))
    
    #뒤 push
    elif comm=='push_back':
        dq.append(int(n[0]))

    #앞 pop
    elif comm=='pop_front':
        print(dq.popleft() if len(dq) >0 else -1)
    
    #뒤 pop
    elif comm=='pop_back':
        print(dq.pop() if len(dq) >0 else -1)
    
    #size
    elif comm=='size':
        print(len(dq))

    #empty
    elif comm=='empty':
        print(1 if len(dq)==0 else 0)
    
    #front
    elif comm == 'front':
        print(dq[0] if len(dq)>0 else -1)
    
    #back
    elif comm=='back':
        print(dq[-1] if len(dq)>0 else -1)